Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -182,9 +182,28 @@ def audit
         display_cop_names:    args.display_cop_names?,
       }.compact
 
+      fetch_and_audit_deps = f.deps.select(&:fetch_and_audit?)
+      fetch_and_audit_deps_installed = fetch_and_audit_deps.select(&:installed?)
+
+      fetch_and_audit_deps.map do |dep|
+        next if fetch_and_audit_deps_installed.include?(dep)
+
+        safe_system ENV["HOMEBREW_BREW_FILE"], "install", dep.name
+      end
+
       fa = FormulaAuditor.new(f, **options)
       fa.audit
 
+      fetch_and_audit_deps.map do |dep|
+        next if fetch_and_audit_deps_installed.include?(dep)
+
+        safe_system ENV["HOMEBREW_BREW_FILE"],
+                    "uninstall",
```

Comment:
Don't think this should be uninstalled here. Let `brew test-bot` handle it instead.

Reply:
What happens if you run `brew audit` locally? You might get a different state on you computer than the one left behind by test-bot. Not sure we want to have half of the logic into test-bot, and the other half in brew.

I make sure in the code to check if brewed curl was installed before, and I uninstall it only if it was not the case.

Reply:
We don't have anywhere else we're doing "install then uninstall" like this outside of `brew test-bot`. If I'm running `audit` multiple times in a row on relevant it's going to be annoying to have it installed and uninstalled every time.